gems/smithy/spec/fixtures/shapes/model.smithy

```diff
@@ -0,0 +1,86 @@
+$version: "2"
+
+namespace smithy.ruby.tests
+
+@paginated(inputToken: "nextToken", outputToken: "nextToken", pageSize: "pageSize")
+service ShapeService {
+    version: "2018-10-31"
+    operations: [
+        Operation
+    ]
+}
+
+@http(method: "POST", uri: "/operation")
+operation Operation {
+    input: OperationInputOutput
+    output: OperationInputOutput
+    errors: [ClientError, ServerError]
+}
+
+@documentation("This is a documentation")
+@xmlName("Structure")
+structure OperationInputOutput {
+    // simple members
+    bigDecimal: BigDecimal
+    bigInteger: BigInteger
+    blob: Blob
+    boolean: Boolean
+    byte: Byte
+    document: Document
+    double: Double
+    enum: Enum
+    float: Float
+    integer: Integer
+    intEnum: IntEnum
+    long: Long
+    short: Short
+    string: String
+    timestamp: Timestamp
+
+    // member with trait
+    @required
+    id: String
+
+    // complex members
+    structure: OperationInputOutput
```

Comment:
This is meant to be recursive right? Its a little weird to use the OperationInputOutput inside the OperationInputOutput and indeed the @output trait forbids it: https://smithy.io/2.0/spec/type-refinement-traits.html#output-trait

Its great to include a recursive shape in here, but I think we may want to not use the Input/Output?

Reply:
Edit: NVM - I see Smithy's kitchen sink does the same thing: https://github.com/smithy-lang/smithy/blob/main/smithy-aws-protocol-tests/model/awsJson1_1/kitchen-sink.smithy#L1035

Reply:
Instead of recursion here, we can use the recursive test model that I added and test it in a separate context. The ensures we always cover the case because if removed from the model, this still passes tests.